Exiled.Events/EventArgs/Scp3114/SlappingPlayerEventArgs.cs

```diff
@@ -0,0 +1,78 @@
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.EventArgs.Scp3114
+{
+    using API.Features;
+    using Interfaces;
+    using PlayerRoles.PlayableScps.Scp3114;
+
+    using Scp3114Role = Exiled.API.Features.Roles.Scp3114Role;
+
+    /// <summary>
+    ///     Contains all information before SCP-3114 slaps a player.
+    /// </summary>
+    // ReSharper disable UnusedAutoPropertyAccessor.Global
+    // ReSharper disable MemberCanBePrivate.Global
+    public sealed class SlappingPlayerEventArgs : IScp3114Event, IDeniableEvent
+    {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="SlappingPlayerEventArgs" /> class.
+        /// </summary>
+        /// <param name="instance">
+        ///     The <see cref="Scp3114Slap"/> instance which this is being instantiated from.
+        /// </param>
+        /// <param name="targetHub">
+        ///     The <see cref="ReferenceHub"/> of the player who was being targeted.
+        /// </param>
+        /// <param name="humeShieldToGive">
+        ///     The amount of hume shield cooldown that is given to the player.
+        /// </param>
+        /// <param name="damageAmount">
+        ///     The amount of damage to do to the target.
+        /// </param>
+        /// <param name="isAllowed">
+        ///     <inheritdoc cref="IDeniableEvent.IsAllowed"/>
+        /// </param>
+        public SlappingPlayerEventArgs(Scp3114Slap instance, ReferenceHub targetHub, float damageAmount, float humeShieldToGive, bool isAllowed = true)
+        {
+            Player = Player.Get(instance.Owner);
+            Target = Player.Get(targetHub);
+            DamageAmount = damageAmount;
+            Scp3114 = Player.Role.As<Scp3114Role>();
+            HumeShieldToGive = humeShieldToGive;
+            IsAllowed = isAllowed;
+        }
+
+        /// <inheritdoc/>
+        /// <summary>
+        ///     The <see cref="Player"/> who is Scp-3114.
+        /// </summary>
+        public Player Player { get; set; }
+
+        /// <inheritdoc cref="IScp3114Event.Scp3114"/>
+        public Scp3114Role Scp3114 { get; }
+
+        /// <summary>
+        ///     Gets the <see cref="Player"/> who is being slapped.
+        /// </summary>
+        public Player Target { get; }
+
+        /// <summary>
+        ///     Gets or sets the hume shield amount that will be regenerated for the Scp 3114 <see cref="Player"/>. This will not have any effect if the event is cancelled.
+        /// </summary>
+        public float HumeShieldToGive { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the amount of damage to deal to the <see cref="Target"/>. This will not have any effect if the event is cancelled.
+        /// </summary>
+        public float DamageAmount { get; set; }
+
+        /// <inheritdoc cref="IDeniableEvent.IsAllowed"/>
```

Comment:
You only need this
        /// <inheritdoc/>

Reply:
Is explicitly typing the xml bad though?

Reply:
It actually throws a stylecop error so im leaving it in

Reply:
strange and no it's was not bad just saying than it's wasn't needed strange than you get an error